std/os.zig

```diff
@@ -3509,20 +1866,566 @@ pub const MProtectError = error{
+        return;
+    }
+    switch (errno(system.access(path, mode))) {
         0 => return,
-        posix.EINVAL => unreachable,
-        posix.EACCES => return error.AccessDenied,
-        posix.ENOMEM => return error.OutOfMemory,
-        else => return unexpectedErrorPosix(err),
+        EACCES => return error.PermissionDenied,
+        EROFS => return error.PermissionDenied,
+        ELOOP => return error.PermissionDenied,
+        ETXTBSY => return error.PermissionDenied,
+        ENOTDIR => return error.FileNotFound,
+        ENOENT => return error.FileNotFound,
+
+        ENAMETOOLONG => return error.NameTooLong,
+        EINVAL => unreachable,
+        EFAULT => unreachable,
+        EIO => return error.InputOutput,
+        ENOMEM => return error.SystemResources,
+        else => |err| return unexpectedErrno(err),
     }
 }
+
+pub const PipeError = error{
+    SystemFdQuotaExceeded,
+    ProcessFdQuotaExceeded,
+    Unexpected,
+};
+
+/// Creates a unidirectional data channel that can be used for interprocess communication.
+pub fn pipe() PipeError![2]fd_t {
+    var fds: [2]i32 = undefined;
+    switch (errno(system.pipe(&fds))) {
+        0 => return fds,
+        EINVAL => unreachable, // Invalid parameters to pipe()
+        EFAULT => unreachable, // Invalid fds pointer
+        ENFILE => return error.SystemFdQuotaExceeded,
+        EMFILE => return error.ProcessFdQuotaExceeded,
+        else => |err| return unexpectedErrno(err),
+    }
+}
+
+pub fn pipe2(flags: u32) PipeError![2]fd_t {
+    var fds: [2]i32 = undefined;
+    switch (errno(system.pipe2(&fds, flags))) {
+        0 => return fds,
+        EINVAL => unreachable, // Invalid flags
+        EFAULT => unreachable, // Invalid fds pointer
+        ENFILE => return error.SystemFdQuotaExceeded,
+        EMFILE => return error.ProcessFdQuotaExceeded,
+        else => |err| return unexpectedErrno(err),
+    }
+}
+
+pub const SysCtlError = error{
+    PermissionDenied,
+    SystemResources,
+    Unexpected,
+};
+
+pub fn sysctl(
+    name: []const c_int,
+    oldp: ?*c_void,
+    oldlenp: ?*usize,
+    newp: ?*c_void,
+    newlen: usize,
+) SysCtlError!void {
+    switch (errno(system.sysctl(name.ptr, name.len, oldp, oldlenp, newp, newlen))) {
+        0 => return,
+        EFAULT => unreachable,
+        EPERM => return error.PermissionDenied,
+        ENOMEM => return error.SystemResources,
+        else => |err| return unexpectedErrno(err),
+    }
+}
+
+pub fn sysctlbynameC(
+    name: [*]const u8,
+    oldp: ?*c_void,
+    oldlenp: ?*usize,
+    newp: ?*c_void,
+    newlen: usize,
+) SysCtlError!void {
+    switch (errno(system.sysctlbyname(name, oldp, oldlenp, newp, newlen))) {
+        0 => return,
+        EFAULT => unreachable,
+        EPERM => return error.PermissionDenied,
+        ENOMEM => return error.SystemResources,
+        else => |err| return unexpectedErrno(err),
+    }
+}
+
+pub fn gettimeofday(tv: ?*timeval, tz: ?*timezone) void {
```

Comment:
Move to `std.time`? Or possibly remove all-together? (on posixy operating systems we should be preferring `clock_gettime`, on windows `GetSystemTimeAsFileTime` or similar)

Reply:
It's called from std.time for the darwin target. I'm not sure why darwin is treated specially; it does seem that clock_gettime should work there too. maybe @tgschultz knows why

Reply:
> I'm not sure why darwin is treated specially; it does seem that clock_gettime should work there too. maybe @tgschultz knows why

From another project I work on https://github.com/wahern/cqueues/blob/36b4b8cf1d8d3e3224cab04357dda3a633c7f0ff/src/cqueues.c#L128 : 
>  * OS X didn't implement the clock_gettime() POSIX interface until macOS
> * 10.12 (Sierra). But it did provide a monotonic clock through
> * mach_absolute_time(). On i386 and x86_64 architectures this clock is in
> * nanosecond units, but not so on other devices. mach_timebase_info()
> * provides the conversion parameters.

Reply:
daurnimator's info is indeed why I didn't just call clock_gettime. I thought I left a comment about it in the code.